src/models/converters/data_decoded.rs

```diff
@@ -13,12 +13,13 @@ impl DataDecoded {
     pub(super) async fn to_settings_info(
         &self,
         info_provider: &impl InfoProvider,
+        chain_id: &str,
     ) -> Option<SettingsInfo> {
         match self.method.as_str() {
             SET_FALLBACK_HANDLER => {
                 let handler = self.get_parameter_single_value_at(0)?;
                 Some(SettingsInfo::SetFallbackHandler {
-                    handler_info: info_provider.contract_info(&handler).await.ok(),
+                    handler_info: info_provider.contract_info(chain_id, &handler).await.ok(),
```

Comment:
Should we initialize the provider with the chain id? this way we don'T have to pass it around

Reply:
ok I will try that.

Reply:
I will move this PR back to draft as this change changes substantially how I did everything. I think it would even simplify the macros

Reply:
Having a struct like so:

```rust
pub struct DefaultInfoProvider<'p, C: Cache> {
    pub chain_id: &'p str,
    client: &'p reqwest::Client,
    cache: &'p C,
    // Mutex is an async Mutex, meaning that the lock is non-blocking
    safe_cache: Mutex<HashMap<String, Option<SafeInfo>>>,
    token_cache: Mutex<HashMap<String, Option<TokenInfo>>>,
    chain_cache: Mutex<HashMap<String, Option<ChainInfo>>>,
}
```

Will allow to simplify the `core_uri!` macros further, we don't need to pass the `chain_id` as it is part of the `info_provider`